### DISCUSSION TITLE
Unsupported format: %mp3 in liquidsoap 2.04 Windows

### DISCUSSION THREAD

**Discussion by tobycm** [savonet/liquidsoap#2465] [User Support]
**Describe the bug**
I tried to use %mp3 to output for icecast but the error is:
```
Error 12: Unsupported format: %mp3(stereo,bitrate=128,samplerate=44100,id3v2=false).
You must be missing an optional dependency.
```
This works fine on Linux Ubuntu 20.04 with the same liquidsoap version.
I believe I'm missing dependencies but I don't know how to get them. Liquidsoap installation is on PATH

**To Reproduce**
My script:
```
pl = playlist("./playlist.txt")

# stream the playlist
output.icecast(%mp3,
host = host, 
port = port, 
password = password, 
mount = station_name,
source)
```

**Expected behavior**
Start to stream just fine

**Version details**
 - OS: Windows 10 Version 21H2 Build 19044.1706
 - Version 2.04

**Install method**
Via Github release assets

**Comment by toots** [comment]
Hi,

We need to find a way to document this better but the windows build only has the `%ffmpeg` encoder enabled. There is some build conflicts when including all encoder library in the windows binary (because it is statically built) and this was decided because the `%ffmpeg` encoder already provides the same functionalities as all the other encoders (and more!).

**Reply by smimram** [reply]
Maybe could we implement %mp3 with ffmpeg when lame is not present?

**Comment by tobycm** [comment]
Hi,
I have made a new format for Liquidsoap to use. It is:
```
%ffmpeg(
    format = "mp3",
    %audio(
        codec = "mp3"
    ))
```

FFMPEG reports mp3 both in `ffmpeg -formats` and `ffmpeg -codecs`
But Liquidsoap still complaining about not finding mp3 encoder:
```
Cannot find encoder mp3: The requested ffmpeg encoder was not found, please make sure that ffmpeg was compiled with support for it.
```

I tried `ffmpeg -i a_video_file_with_music.mp4 -acodec mp3 -f mp3 output_audio_file.mp3` and it works and exports the audio file just fine

**Reply by Russsgithub** [reply]
Hi @tobycm , try it with the codec as 'libmp3lame' which is the mp3 codec.

This is from the docs
```
%ffmpeg(format="mp3",
  %audio(codec="libmp3lame"),
  %video.copy)
```

https://www.liquidsoap.info/doc-dev/ffmpeg_encoder.html

**Reply by tobycm** [reply]
Thank you!

**Comment by vannan5** [comment]
radio0=input.portaudio()
output.icecast(%ffmpeg(format="mp3",%audio(codec="libshine",samplerate=48000)),
host="xyz.com",
port=80,
password="xyz",
mount="test",
radio0
)

Error 8: Unsupported format: %ffmpeg(format="mp3", %audio(codec="libshine", samplerate=48000))
